tools/azure-pipelines-tasks-json-generate.mjs

```diff
@@ -0,0 +1,59 @@
+import os from 'os';
+import { promisify } from 'util';
+import fs from 'fs-extra';
+import g from 'glob';
+import JSON5 from 'json5';
+import shell from 'shelljs';
+import Git from 'simple-git';
+import path from 'upath';
+
+const glob = promisify(g);
+
+const localPath = path.join(os.tmpdir(), 'azure-pipelines-tasks');
+
+await (async () => {
+  await fs.ensureDir(localPath);
+  const git = Git(localPath);
+
+  if (await git.checkIsRepo()) {
+    await git.pull();
```

Comment:
There's some chance that a local git repo could get into an unpullable state, e.g. if someone force pushed to the origin? But if this doesn't affect our automated systems then it's no big deal and can be resolved by the user through manual deletion

Reply:
I think it's unlikely that someone will do that, given that it's a protected branch that requires PRs. Like you said, not an issue for automated systems, and can easily be resolved locally.